### PR TITLE
Make routes compile all html files to index.html

### PR DESCRIPTION
**Pull request by lifeiscontent**
This felt more natural to me, what do you think @iansinnott ?

### CODE REVIEW

**Comment by iansinnott** [comment]
Ah, cool. Thanks for the PR. I didn't see this before responding to your other comment. Will review.

File: src/utils.js [hunk 1]
```diff
@@ -231,7 +231,7 @@ export const getAssetKey = (location: string): string => {
   } else if (basename === '*') {
     filename = '404.html';
   } else {
-    filename = basename + '.html';
+    filename = basename + '/index.html';
```
Comment (iansinnott) [inline src/utils.js]: Nice and simple, i like it :)

**Comment by iansinnott** [comment]
The updates look clean. Nice job updating all the tests. My main concern with this is how it might effect the [react-static-boilerplate](https://github.com/iansinnott/react-static-boilerplate). I bet everything would just work, but it will take some manual qa testing.

**Comment by lifeiscontent** [comment]
@iansinnott I found the real bug, originally you had the option documented to be able to add a trailing slash to your routes for it to generate an index.html version of a file. I totally agree with that contract, so I'll update this PR accordingly it just turns out that you didn't write a test case to cover this edge case, but it IS documented.

**Comment by lifeiscontent** [comment]
@iansinnott I might make a few more proposals in the coming days. Just a heads up 👍

**Comment by lifeiscontent** [comment]
Also, thanks for getting back to me so quickly 💯

**Comment by lifeiscontent** [comment]
@iansinnott just for reference, here is where you documented the spec: https://github.com/ConceptualCode/react-static-webpack-plugin/blob/30f3ce1bcf672e97ae9874410919a907468bf655/src/utils.js#L213

**Comment by lifeiscontent** [comment]
@iansinnott I'd really love for you to take a look at this issue too #22

**Comment by iansinnott** [comment]
Ah, nice find! I myself had completely forgotten about that convention since I hadn't made use of it.

So are you feeling good about this MR? Looks like all the tests pass. Also thanks for adding a new test to cover this case.

**Comment by lifeiscontent** [comment]
@iansinnott This solves all of the problems I have currently, and it also follows your documentation. So if its okay with you, I'd say its good to merge! 👍

**Comment by iansinnott** [comment]
Great, I'm going to cut a new release to get this on NPM.
